### PR TITLE
WRP-22351: Publish the ColorPicker component

### PR DESCRIPTION
**Pull request by ion-andrusciac-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Added a new private component named ColorPicker

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Helper functions were moved into the `utils.js` file and added unit-tests for the component

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-22351

### Comments

Enact-DCO-1.0-Signed-off-by: Ion Andrusciac ion.andrusciac@lgepartner.com

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ The following is a curated list of changes in the Enact sandstone module, newest
 
 ### Added
 
+- `sandstone/ColorPicker` component
```
Comment (daniel-stoian-lgp) [inline CHANGELOG.md:9]: we don't need changelog for private components

File: ColorPicker/utils.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/**
+ * Checks for valid hex color, otherwise returns the black color.
+ *
+ * @param {String} hex	Hex color
+ * @returns {String}	Valid hex color
```
Comment (daniel-stoian-lgp) [inline ColorPicker/utils.js:5]:
```suggestion
 * @returns {String} Valid hex color
```

**Review by daniel-stoian-lgp — COMMENTED** [review]
just a few comments

**Review by daniel-stoian-lgp — APPROVED** [review]

File: ColorPicker/ColorPicker.module.less [hunk 3]
```diff
@@ -0,0 +1,93 @@
+// ColorPicker.module.less
+//
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPicker.module.less:2]: we can remove `//` here

File: ColorPicker/ColorPicker.module.less [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+		.colorSlider {
+			margin: 0 27px;
+		}
+
+		.colorSliderText {
+			font-size: 54px;
+			margin: 0 24px;
+		}
+	}
+}
```
Comment (adrian-cocoara-lgp) [inline ColorPicker/ColorPicker.module.less:93]: add an empty line

File: samples/sampler/stories/qa/ColorPicker.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import ColorPicker, {ColorPickerBase} from 'ColorPicker';
```
Comment (0x64) [inline samples/sampler/stories/qa/ColorPicker.js:1]:
```suggestion
import ColorPicker, {ColorPickerBase} from '@enact/sandstone/ColorPicker';
```
QA Sampler does not run without this change.

**Comment by 0x64** [comment]
The newly added QA sample is "story" indicated by bookmark icon (just like the 'default' sampler), but other samples in QA sampler shows stories inside other level. Could you please set the level of a story? You can refer other QA sampler samples.
Note: two samplers have different levels from each other.

File: samples/sampler/stories/qa/ColorPicker.js (hunk 5 above)
Comment (ion-andrusciac-lgp) [reply samples/sampler/stories/qa/ColorPicker.js]: @0x64 I pushed the changes, could you review them again, please?

Comment (0x64) [reply samples/sampler/stories/qa/ColorPicker.js]: @ion-andrusciac-lgp I checked updated changes. Thank you!

**Pull request merged**